src/test/scala/is/hail/expr/ir/PruneSuite.scala

```diff
@@ -316,6 +316,19 @@ class PruneSuite extends SparkSuite {
       Array(subsetMatrixTable(mat.typ, "g.e2", "sa.c2")))
   }
 
+  @Test def testTableToMatrixTableMemo() {
+    val ttmt = TableToMatrixTable(tab,
+      rowKey = FastIndexedSeq("1"),
+      colKey = FastIndexedSeq("2"),
+      rowFields = FastIndexedSeq(),
+      colFields = FastIndexedSeq("4"),
+      partitionKey = FastIndexedSeq("1")
+    )
+
+    checkMemo(ttmt,
+      subsetMatrixTable(ttmt.typ, "sa.4.A"), Array(subsetTable(tab.typ, "row.1", "row.2.2A", "row.4.A")))
```

Comment:
It's not clear to me how this test is working. The way I read this is the subsetMatrixTable type such equal the subsetTable type. But you're not selecting for `row.2.2A` anywhere. I would have expected `row.2`. `row.4.A` and `row.1` make sense to me.

Reply:
all IR nodes have full dependence on all keys, and row.2 is becoming a column key so we need full (row.2.2A) dependence. If I said just `row.2` here, then it would depend on the empty struct rather than the struct with 2A in it.

Reply:
this stuff could possibly be written a bit more clearly with extra code that says `full("row.1")`, `full("row.2")`, `minimal("row.4.A")` but that seems like a pain

Reply:
but you point out an important problem, which is that only Cotton (who reviewed it) and I (who wrote it) understand what this code is supposed to do and how it does that.